pkg/validations/upgradevalidations/eksaversions_test.go

```diff
@@ -0,0 +1,223 @@
+				test.Expect(err.Error()).To(ContainSubstring(tc.wantErr.Error()))
+			} else {
+				test.Expect(err).To(BeNil())
+			}
+		})
+	}
+}
+
+func TestValidateEKSAVersionSkewFailGetCluster(t *testing.T) {
+	tt := newTest(t, withKubectl())
+	wantErr := "failed to reach cluster"
+	clusterName := "upgrade-cluster"
+	uCluster := getCluster(clusterName)
+
+	ctx := context.Background()
+	tt.kubectl.EXPECT().GetEksaCluster(ctx, uCluster, uCluster.Name).Return(nil, errors.New(wantErr))
+
+	err := upgradevalidations.ValidateEKSAVersionSkew(ctx, "", tt.kubectl, uCluster)
+	tt.Expect(err.Error()).To(Equal(wantErr))
+}
+
+func TestValidateEKSAVersionSkewFailBundleRef(t *testing.T) {
+	tt := newTest(t, withKubectl())
+	wantErr := "cluster bundlesRef cannot be nil"
+	clusterName := "upgrade-cluster"
+	uCluster := getCluster(clusterName)
+	upgradeCluster := anywhereCluster(clusterName)
+
+	upgradeCluster.Spec.BundlesRef = nil
+	ctx := context.Background()
+	tt.kubectl.EXPECT().GetEksaCluster(ctx, uCluster, uCluster.Name).Return(upgradeCluster, nil)
+
+	err := upgradevalidations.ValidateEKSAVersionSkew(ctx, "", tt.kubectl, uCluster)
+	tt.Expect(err.Error()).To(Equal(wantErr))
+}
+
+func TestValidateEKSAVersionSkewFailParseTags(t *testing.T) {
+	tt := newTest(t, withKubectl())
+	wantErr := "could not find tag in Eksa Cluster Controller Image"
+	clusterName := "upgrade-cluster"
+	uCluster := getCluster(clusterName)
+	upgradeCluster := anywhereCluster(clusterName)
+	upgradeCluster.Spec.BundlesRef = &anywherev1.BundlesRef{
+		Name:      "bundles-1",
+		Namespace: constants.EksaSystemNamespace,
+	}
+	uri := "badvalue"
+	vb := releasev1alpha1.VersionsBundle{
+		Eksa: releasev1alpha1.EksaBundle{
+			ClusterController: releasev1alpha1.Image{
+				URI: uri,
+			},
+		},
+	}
+	bundle := &releasev1alpha1.Bundles{
+		Spec: releasev1alpha1.BundlesSpec{
+			VersionsBundles: []releasev1alpha1.VersionsBundle{
+				vb,
+			},
+		},
+	}
+
+	ctx := context.Background()
+	tt.kubectl.EXPECT().GetEksaCluster(ctx, uCluster, uCluster.Name).Return(upgradeCluster, nil)
+	tt.kubectl.EXPECT().GetBundles(ctx, uCluster.KubeconfigFile, upgradeCluster.Spec.BundlesRef.Name, upgradeCluster.Spec.BundlesRef.Namespace).Return(bundle, nil)
+
+	err := upgradevalidations.ValidateEKSAVersionSkew(ctx, "", tt.kubectl, uCluster)
+	tt.Expect(err.Error()).To(Equal(wantErr))
+}
+
+func getCluster(name string) *types.Cluster {
+	return &types.Cluster{
+		Name: name,
+	}
+}
+
+func anywhereCluster(name string) *anywherev1.Cluster {
+	return &anywherev1.Cluster{
+		ObjectMeta: v1.ObjectMeta{
+			Name: name,
+		},
+		Spec: anywherev1.ClusterSpec{
+			ManagementCluster: anywherev1.ManagementCluster{
+				Name: name,
+			},
+		},
+	}
+}
+
+type clusterTest struct {
+	*WithT
+	kubectl     *mocks.MockKubectlClient
+	provider    *providermocks.MockProvider
+	clusterSpec *cluster.Spec
+}
+
+type clusterTestOpt func(t *testing.T, ct *clusterTest)
+
+func newTest(t *testing.T, opts ...clusterTestOpt) *clusterTest {
```

Comment:
Every test seems to use `withKubectl()`. I would imagine it'd be simpler to have `newTest(t *testing.T)`